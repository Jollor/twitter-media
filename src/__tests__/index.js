jest.mock('../api-client');

const TwitterMedia = require('../');
const path = require('path');
const fs = require('fs');

const image = fs.readFileSync(path.join(__dirname, '..', '..', 'assets', 'image.jpg'));
const video = fs.readFileSync(path.join(__dirname, '..', '..', 'assets', 'video.mp4'));

describe('Promise API', () => {
    it('Image upload', () => {
        const twitter = new TwitterMedia();
        return expect(twitter.uploadMedia('image', image)).resolves.toMatchSnapshot();
    });

    it('Image-set upload', () => {
        const twitter = new TwitterMedia();
        return expect(twitter.uploadImageSet([image, image, image])).resolves.toMatchSnapshot();
    });

    it('Video upload', () => {
        const twitter = new TwitterMedia();
        return expect(twitter.uploadMedia('video', video)).resolves.toMatchSnapshot();
    });
});

describe('Callback API', () => {
    it('Image upload', () => {
        const twitter = new TwitterMedia();

        expect.assertions(1);
        twitter.uploadMedia('image', image, (error, mediaID, body) => {
            expect(body).toMatchSnapshot();
        });
    });

    it('Image-set upload', () => {
        const twitter = new TwitterMedia();

        expect.assertions(1);
        twitter.uploadImageSet([image, image, image], (error, body) => {
            expect(body).toMatchSnapshot();
        });
    });

    it('Video upload', () => {
        const twitter = new TwitterMedia();

        expect.assertions(1);
        twitter.uploadMedia('video', video, (error, mediaID, body) => {
            expect(body).toMatchSnapshot();
        });
    });
});
