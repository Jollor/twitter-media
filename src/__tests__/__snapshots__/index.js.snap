// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Callback API Image upload 1`] = `
Object {
  "expires_after_secs": 86400,
  "image": Object {
    "h": 0,
    "image_type": "image/jpeg",
    "w": 0,
  },
  "media_id": 1,
  "media_id_string": "1",
  "size": 55984,
}
`;

exports[`Callback API Image-set upload 1`] = `
Array [
  Object {
    "expires_after_secs": 86400,
    "image": Object {
      "h": 0,
      "image_type": "image/jpeg",
      "w": 0,
    },
    "media_id": 1,
    "media_id_string": "1",
    "size": 55984,
  },
  Object {
    "expires_after_secs": 86400,
    "image": Object {
      "h": 0,
      "image_type": "image/jpeg",
      "w": 0,
    },
    "media_id": 2,
    "media_id_string": "2",
    "size": 55984,
  },
  Object {
    "expires_after_secs": 86400,
    "image": Object {
      "h": 0,
      "image_type": "image/jpeg",
      "w": 0,
    },
    "media_id": 3,
    "media_id_string": "3",
    "size": 55984,
  },
]
`;

exports[`Callback API Video upload 1`] = `
Object {
  "expires_after_secs": 86399,
  "media_id": 1,
  "media_id_string": "1",
  "processing_info": Object {
    "progress_percent": 100,
    "state": "succeeded",
  },
  "size": 6295234,
  "video": Object {
    "video_type": "video/mp4",
  },
}
`;

exports[`Promise API Image upload 1`] = `
Object {
  "expires_after_secs": 86400,
  "image": Object {
    "h": 0,
    "image_type": "image/jpeg",
    "w": 0,
  },
  "media_id": 1,
  "media_id_string": "1",
  "size": 55984,
}
`;

exports[`Promise API Image-set upload 1`] = `
Array [
  Object {
    "expires_after_secs": 86400,
    "image": Object {
      "h": 0,
      "image_type": "image/jpeg",
      "w": 0,
    },
    "media_id": 1,
    "media_id_string": "1",
    "size": 55984,
  },
  Object {
    "expires_after_secs": 86400,
    "image": Object {
      "h": 0,
      "image_type": "image/jpeg",
      "w": 0,
    },
    "media_id": 2,
    "media_id_string": "2",
    "size": 55984,
  },
  Object {
    "expires_after_secs": 86400,
    "image": Object {
      "h": 0,
      "image_type": "image/jpeg",
      "w": 0,
    },
    "media_id": 3,
    "media_id_string": "3",
    "size": 55984,
  },
]
`;

exports[`Promise API Video upload 1`] = `
Object {
  "expires_after_secs": 86399,
  "media_id": 1,
  "media_id_string": "1",
  "processing_info": Object {
    "progress_percent": 100,
    "state": "succeeded",
  },
  "size": 6295234,
  "video": Object {
    "video_type": "video/mp4",
  },
}
`;
